src/placeos-models/api_key.cr

```diff
@@ -0,0 +1,106 @@
+require "random"
+require "crypto/bcrypt"
+
+require "./base/model"
+require "./user_jwt"
+
+module PlaceOS::Model
+  class ApiKey < ModelBase
+    include RethinkORM::Timestamps
+
+    table :api_key
+
+    attribute name : String, es_subfield: "keyword"
+    attribute description : String = ""
+    attribute scopes : Array(String) = ["public"]
+
+    # when nil it defaults to the users permissions
+    attribute permissions : UserJWT::Permissions? = nil
+
+    attribute secret : String = ->{ Random::Secure.urlsafe_base64(32) }, mass_assignment: false
+
+    belongs_to User
+    belongs_to Authority
+
+    def user=(user)
+      super(user)
+      self.authority_id = user.authority_id
+    end
```

Comment:
Due to some weirdness with active-model, I think this is the way to do overrides of getters/setters

https://github.com/PlaceOS/models/blob/master/src/placeos-models/authority.cr#L27-L35